crates/router/src/core/files.rs

```diff
@@ -0,0 +1,104 @@
+#[cfg(feature = "s3")]
+pub mod s3_utils;
+pub mod transformers;
+
+#[cfg(not(feature = "s3"))]
+pub mod fs_utils;
+
+use api_models::files;
+use error_stack::{IntoReport, ResultExt};
+
+use super::errors::{self, RouterResponse};
+use crate::{
+    consts,
+    routes::AppState,
+    services::{self, ApplicationResponse},
+    types::{api, storage},
+};
+
+pub async fn files_create_core(
+    state: &AppState,
+    merchant_account: storage::merchant_account::MerchantAccount,
+    create_file_request: api::CreateFileRequest,
+) -> RouterResponse<files::CreateFileResponse> {
+    transformers::validate_file_upload(
+        state,
+        merchant_account.clone(),
+        create_file_request.clone(),
+    )
```

Comment:
it's better if we use a crate (for example, any crate that provides bindings to the libmagic C library) and find the Mime type of the files by ourselves in here since there's nothing that prevents the merchant from sending the wrong Content-Type in the request.

Accordingly, if we're going to infer file mime types, it would be a good idea to add the corresponding file extension at the end of the S3 key for better differentiation in S3 and just in case some connectors are sensitive to that.